x-pack/plugins/licensing/server/plugin.ts

```diff
@@ -99,32 +111,12 @@ export class LicensingPlugin implements Plugin<LicensingPluginSetup, LicensingPl
     this.logger.debug('Setting up Licensing plugin');
     const pollingFrequency = this.config.api_polling_frequency;
 
-    async function callAsInternalUser(
-      ...args: Parameters<ILegacyScopedClusterClient['callAsInternalUser']>
-    ): ReturnType<ILegacyScopedClusterClient['callAsInternalUser']> {
-      const [coreStart] = await core.getStartServices();
-      const client = coreStart.elasticsearch.legacy.client;
-      return await client.callAsInternalUser(...args);
-    }
-
-    const client: ILegacyClusterClient = {
-      callAsInternalUser,
-      asScoped(request?: ScopeableRequest): ILegacyScopedClusterClient {
-        return {
-          async callAsCurrentUser(
-            ...args: Parameters<ILegacyScopedClusterClient['callAsCurrentUser']>
-          ): ReturnType<ILegacyScopedClusterClient['callAsCurrentUser']> {
-            const [coreStart] = await core.getStartServices();
-            const _client = coreStart.elasticsearch.legacy.client;
-            return await _client.asScoped(request).callAsCurrentUser(...args);
-          },
-          callAsInternalUser,
-        };
-      },
-    };
+    const clientPromise = core.getStartServices().then(([{ elasticsearch }]) => {
+      return elasticsearch.client;
+    });
 
     const { refresh, license$ } = this.createLicensePoller(
-      client,
+      clientPromise,
```

Comment:
It seems we can remove `createLicensePoller` from `setup` contract. Monitoring uses a version from `start` contract https://github.com/elastic/kibana/blob/fd524e1066e0aaeae2de61e4df3db432ae90bc99/x-pack/plugins/monitoring/server/plugin.ts#L198

Not sure about other API, but maybe we can remove them too. In a follow-up maybe.